docs/articles/blog/2020-7-15-introducing-the-testcafe-jenkins-plugin.md

```diff
@@ -0,0 +1,170 @@
+
+![Jenkins plugin search UI](/testcafe/images/jenkins-plugin/plugin-search-ui.png)
+
+### Step 3. Install the necessary Node packages
+
+Install the main testcafe library, as well as the [testcafe-reporter-jenkins](https://www.npmjs.com/package/testcafe-reporter-jenkins) package. The latter is needed to generate Jenkins-friendly JUnit format reports.
+
+To do this, add the following *build step* to your Jenkins project:
+
+```bash
+npm i testcafe testcafe-reporter-jenkins
+```
+
+### Step 4. Configure the tests
+
+You can configure your tests via:
+
+* the [command line interface](https://devexpress.github.io/testcafe/documentation/reference/command-line-interface.html)
+* the JavaScript/TypeScript API
+* the .testcaferc.json [configuration file](https://devexpress.github.io/testcafe/documentation/reference/configuration-file.html)
+
+The last option provides an easy, declarative way to define your test settings. You can store this file in the same folder as your tests.  Below is the configuration we're going to use in this tutorial:
+
+```js
+{
+    "browsers": [
+        "chromium:headless:emulation:device=iPhone 4",
+        "chromium:headless:emulation:device=iPad Mini",
+        "chromium:headless"
+    ],
+    "src": "test/e2e/**/*",
+    "screenshots": {
+        "takeOnFails": true,
+        "fullPage": true
+    },
+    "videoPath": "videos",
+    "reporter": {
+        "name": "jenkins",
+        "output": "report.xml"
+    },
+    "appCommand": "node server.js"
+}
+```
+
+The *Browser* array contains the list of browsers TestCafe will use to run our test. The [TestCafe browser syntax](https://devexpress.github.io/testcafe/documentation/guides/concepts/browsers.html#use-chromium-device-emulation) allows us to specify the mobile devices we want Chromium to emulate. 
+
+The *name* property of the *reporter* object is set as *jenkins*, which ensures that the reports generated by TestCafe can be properly parsed by the server.
+
+If you decide against using a config file, don't forget to [manually set](https://devexpress.github.io/testcafe/documentation/reference/command-line-interface.html#-r-nameoutput---reporter-nameoutput) the report format when launching your tests, like so:
+
+```bash
+testcafe chrome test.js -r jenkins:report.xml
+```
+
+Otherwise, just run the following command:
+
+```bash
+npx testcafe
+```
+
+### Step 5. Configure the reporter
+
+Turn on the [JUnit](https://plugins.jenkins.io/junit/) reporter plugin: add the "Publish JUnit test result report" *post-build action*. To display screenshots and videos alongside your test results, select the option "Include links to TestCafe artifacts" from the "Additional test report features" drop-down.
+
+Important: Don't forget to check the "Retain long standard output/error" box. Otherwise the plugin will automatically truncate information concerning test screenshots and videos.
```

Comment:
```suggestion
Important: Remember to check the "Retain long standard output/error" box. Otherwise, the plugin will automatically truncate information about test screenshots and videos.
```

Reply:
I probably need to rewrite the second sentence. Thing is: Jenkins reports generated without this option enabled feature incomplete (and, therefore, useless) URLs. It's not the metadata that gets truncated, as the current phrasing (`information about test screenshots`) would have you believe, but the most important part of the actual data.

Would `Otherwise, the reporter plugin will automatically truncate the URLs of your videos and screenshots.` work?

Reply:
`Otherwise, the reporter plugin will automatically truncate the URLs of your videos and screenshots.`

Sounds good. Please apply the change.